dali/ft/filecopy.cpp

```diff
@@ -3249,13 +3256,117 @@ void FileSprayer::updateTargetProperties()
                 if (stricmp(aname, "Protect") != 0)
                     curProps.addPropTree(aname, createPTreeFromIPT(&iter->query()));
             }
+
+            // Handle history
+            IPropertyTree *curHistory = lock.queryHistory();
+
+            if (srcHistory)
+            {
+                // copy history records from source file
+                Owned<IPropertyTreeIterator> historyIter = srcHistory->getElements("*");
+                ForEach(*historyIter)
+                {
+                    const char *aname = historyIter->query().queryName();
+                    curHistory->addPropTree(aname, createPTreeFromIPT(&historyIter->query()));
+                }
+            }
+
+            // Add new record about this operation
+            IPropertyTree * newRecord = createPTree("Origin");
+
+            CDateTime temp;
+            temp.setNow();
+            unsigned hour, min, sec, nanosec;
+            temp.getTime(hour, min, sec, nanosec);
+            temp.setTime(hour, min, sec, 0);
+
+            StringBuffer timestr;
+            newRecord->addProp("@timestamp",temp.getString(timestr).str());
+
+            StringBuffer val;
+            srcAttr->getProp("@owner", val);
+            newRecord->addProp("@owner", val.str());
+
+            srcAttr->getProp("@workunit", val.clear());
+            if (0 == val.length())
+                val.set("N/A");
```

Comment:
If something not there which usually should be then I can feel there is a bug somewhere. So in case of Spray there is no previous workunit and originally it filled with and empty string, which can be suspicious as well. That's why I store that "N/A" string.
Moreover if some filed missing, than it can cause further headache when the history would be export as CSV.